invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -0,0 +1,108 @@
+
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class CSLCreatorSchema(Schema):
+    """Creator/contributor common schema."""
+
+    family = fields.Str(attribute="person_or_org.name")
+
+
+class CSLJSONSchema(Schema):
+    """CSL Marshmallow Schema."""
+
+    id_ = SanitizedUnicode(data_key="id", attribute='id')
+    type_ = fields.Method('get_type', data_key="type")
+    title = SanitizedUnicode(attribute="metadata.title")
+    abstract = SanitizedUnicode(attribute="metadata.description")
+    author = fields.List(
+        fields.Nested(CSLCreatorSchema), attribute='metadata.creators')
+    issued = fields.Method('get_issued')
+    language = fields.Method('get_language')
+    version = SanitizedUnicode(attribute="metadata.version")
+    note = fields.Method('get_note')
+    doi = fields.Method('get_doi', data_key="DOI")
+    isbn = fields.Method('get_isbn', data_key="ISBN")
+    issn = fields.Method('get_issn', data_key="ISSN")
+    publisher = SanitizedUnicode(attribute='metadata.publisher')
+
+    def _read_resource_type(self, id_):
+        """Retrieve resource type record using service."""
+        return vocabulary_service.read(
+            ('resource_types', id_),
+            system_identity
+        )._record
+
+    def get_type(self, obj):
+        """Get resource type."""
+        resource_type = obj["metadata"]["resource_type"]
+        resource_type_record = self._read_resource_type(resource_type["id"])
+        props = resource_type_record["props"]
+        return props.get("csl", "article")  # article is CSL "Other"
+
+    def get_issued(self, obj):
+        """Get issued dates."""
+        date_parts = []
+        publication_date = obj["metadata"]["publication_date"].split("/")
+        for dates in publication_date:
+            date_parts.append(dates.split("-"))
+
+        return {"date-parts": date_parts}
+
+    def get_language(self, obj):
+        """Get language."""
+        metadata = obj["metadata"]
+        languages = metadata.get("languages")
+
+        return languages[0]["id"] if languages else missing
+
+    def get_doi(self, obj):
+        """Get DOI."""
+        return obj["pids"].get("doi", {}).get("identifier", missing)
+
+    def get_isbn(self, obj):
+        """Get ISBN."""
+        identifiers = obj["metadata"]["identifiers"]
+        for identifier in identifiers:
+            if identifier["scheme"] == "ISBN":
+                return identifier["identifier"]
+
+        return missing
+
+    def get_issn(self, obj):
+        """Get ISSN."""
+        identifiers = obj["metadata"]["identifiers"]
+        for identifier in identifiers:
+            if identifier["scheme"] == "ISSN":
+                return identifier["identifier"]
+
+        return missing
+
+    def get_note(self, obj):
```

Comment:
Question: is it correct to get the note from the funding field?

Reply:
That's one of the open questions above:

> In Zenodo the note contains the funding reference. However, here we have to reconstruct it from `funding.funders` and `funding.awards`... Should we take both into account or just the first (i.e. funders)

We are not sure, but I guess we can treat that as a separate issue